src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -111,74 +154,164 @@ internal abstract class AbstractAddParameterCodeFixProvider<
+            RegisterFixForMethodOverloads(context, arguments, candidates);
+        }
 
-            var comparer = syntaxFacts.StringComparer;
-            var constructorsAndArgumentToAdd = ArrayBuilder<(IMethodSymbol constructor, TArgumentSyntax argument, int index)>.GetInstance();
+        private void RegisterFixForMethodOverloads(CodeFixContext context, SeparatedSyntaxList<TArgumentSyntax> arguments, ImmutableArray<(IMethodSymbol method, TArgumentSyntax argument, int index, TTypeSyntax typeArgument)> methodsAndArgumentToAdd)
+        {
+            // Order by the furthest argument index to the nearest argument index.  The ones with
+            // larger argument indexes mean that we matched more earlier arguments (and thus are
+            // likely to be the correct match).
+            foreach (var (method, argumentToAdd, _, typeArgument) in methodsAndArgumentToAdd.OrderByDescending(t => t.index))
+            {
+                var parameters = method.Parameters.Select(p => p.ToDisplayString(SimpleFormat));
+                var signature = $"{method.Name}({string.Join(", ", parameters)})";
+
+                var title = string.Format(FeaturesResources.Add_parameter_to_0, signature);
+
+                context.RegisterCodeFix(
+                    new MyCodeAction(title, c => FixAsync(context.Document, method, argumentToAdd, arguments, typeArgument, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        private ImmutableArray<(IMethodSymbol method, TArgumentSyntax argument, int index, TTypeSyntax typeArgument)> GetArgumentInsertPositionAndTypeArgumentForMethodCandidates(
+            SemanticModel semanticModel,
+            ISyntaxFactsService syntaxFacts,
+            ImmutableArray<IMethodSymbol> methodCandidates,
+            SeparatedSyntaxList<TArgumentSyntax> arguments,
+            TArgumentSyntax argumentOpt,
+            SeparatedSyntaxList<TTypeSyntax> typeArguments)
+        {
+            var methodsAndArgumentToAdd = ArrayBuilder<(IMethodSymbol constructor, TArgumentSyntax argument, int index, TTypeSyntax typeArgument)>.GetInstance();
 
-            foreach (var constructor in type.InstanceConstructors.OrderBy(m => m.Parameters.Length))
+            foreach (var method in methodCandidates.OrderBy(m => m.Parameters.Length))
             {
-                if (constructor.IsNonImplicitAndFromSource() &&
-                    NonParamsParameterCount(constructor) < arguments.Count)
+                if (method.IsNonImplicitAndFromSource())
                 {
-                    var argumentToAdd = DetermineFirstArgumentToAdd(
-                        semanticModel, syntaxFacts, comparer, constructor,
-                        arguments, argumentOpt);
+                    var argumentToAdd = GetArgumentInsertPositionForMethodCandidate(semanticModel, syntaxFacts, method, arguments, argumentOpt);
```

Comment:
method is called "GetArgumentInsertPoistion".  But variable is called "argumentToAdd".  These don't nominally sync up.
plz wrap long lines.